### DISCUSSION TITLE
Is there a way to change the highlight group of an already created extmark?

### DISCUSSION THREAD

**Discussion by Badhi** [neovim/neovim#29215] [Q&A] [answered]
Say that I have already created an extmark with a hl_group assign to it, so that its range is highlighted with the set highlight group colors. If I need to change the hl_group to a different one later, is there an easier way of doing it?

Or do I have to delete the previous extmark and create new one?

**Comment by zeertzjq** [comment, marked as answer]
You can pass an `id` field in the `opts` argument of `nvim_buf_set_extmark`, which will make it modify an existing extmark instead of creating a new one.
